tests/unit/test_query_language.py

```diff
@@ -63,3 +73,41 @@ def test_dataset() -> None:
             rhs=Value(2000),
         ),
     }
+
+
+# The problem: We'd like to test that operations on query language (QL) elements return
+# the correct query model (QM) elements. We like tests that emphasise what is being
+# tested, and de-emphasise the scaffolding. We dislike test code that looks like
+# production code.
+qm_table = SelectPatientTable(
+    name="patients",
+    schema=TableSchema("patients_schema", int_column=int, date_column=date),
+)
+qm_select_int = SelectColumn(source=qm_table, name="int_column")
+qm_select_date = SelectColumn(source=qm_table, name="date_column")
+
+
+def assert_produces(ql_element, qm_element):
+    assert ql_element.qm_node == qm_element
+
+
+class TestIntSeries:
+    def test_le_value(self):
+        assert_produces(
+            IntSeries(qm_select_int) <= 2000,
+            Function.LE(qm_select_int, Value(2000)),
+        )
```

Comment:
It would be lovely to see a test for the `2000 >= IntSeries(qm_select_int)` case as well to try to prove that we're getting value from these tests beyond just repeating what's in the implementation.

Reply:
Added in 3628ab8. I was unsure whether to add a second call to `assert_produces` in `test_le_value` or to add a new test. I added a new test. In the former case, I thought an explanatory comment was necessary; this became the name of the test in the second case (`test_le_value_reverse`).

I think we discussed what to call this functionality: [_The Operator Overloading_][1] chapter of _Fluent Python_ uses _forward_ and _reverse_ (from [_Implementing the arithmetic operations_][2] in the `numbers` documentation). However, the [_Data Model_][3] chapter of the Python documentation uses _reflected_.

[1]: https://learning.oreilly.com/library/view/fluent-python/9781491946237/ch13.html
[2]: https://docs.python.org/3/library/numbers.html#implementing-the-arithmetic-operations
[3]: https://docs.python.org/3/reference/datamodel.html